### PR TITLE
Allow template files to be overriden

### LINKED ISSUE #20: The template files should be able to be overridden from the theme
The `template` function should first check the active theme directory in the path `/{package}/` for a file and use that if it exists.

Eg.

`wp-content/themes/my-theme/wp-alert-bars/alert-bar-countdown.php`

### PR DESCRIPTION
**Pull request by sanjucta**
Resolves #20 

**Steps to Test**
1) Modify composer.json in your local copy of the dbi site to point to the composer path repo containing the wp-alert-bars branch.
2) Create a folder \'wp-alert-bars\' in themes/deliciousbrains.
3) Make a copy of the template files \'alert-bar.php\' or \'alert-bar-countdown.php\' in themes/deliciousbrains/themes/deliciousbrains and make some modifications in the files
4) The templates should be picked up from themes/deliciousbrains/themes/deliciousbrains, so that your modifications are visible in the alert bars.

### CODE REVIEW

File: src/Display.php [hunk 1]
```diff
@@ -90,7 +90,30 @@ public function add_body_class( $classes ) {
...
+		$template_override = locate_template( $this->template_path(). $part . '.php' );
+		$template = $template_override ? $template_override : $this->default_template( $part );
+		include $template;
+	}
+
+
+	/**
+	 * Get the template path.
+	 * 
+	 * @param string $part Part filename without the extension
```
Comment (rosswintle) [inline src/Display.php:102]: This method does not have a parameter. Please remove `@param`.

File: src/Display.php [hunk 2]
```diff
@@ -90,7 +90,30 @@ public function add_body_class( $classes ) {
...
+	 * @return string
+	 */
+	private function template_path() {
+		return apply_filters( 'wp_alert_bars_template_path', 'wp-alert-bars/' );
```
Comment (polevaultweb) [inline src/Display.php:106]: This should be `dbi_wp_alert_bars_template_path`

File: templates/alert-bar.php [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 <div id="alert-bar-<?php echo $alert_bar->id; ?>" class="alert-bar<?php echo $class; ?>" data-slug="<?php echo $alert_bar->key; ?>-alert-bar" style="display: none;">
 	<?php if ( $alert_bar->countdown_end ) {
-		include_once 'alert-bar-countdown.php';
+		$this->template( 'alert-bar-countdown', array( 'alert_bar' => $alert_bar ) );
```
Comment (rosswintle) [inline templates/alert-bar.php:20]: Style thing: Our standard is now to use short array syntax.
```suggestion
		$this->template( 'alert-bar-countdown', [ 'alert_bar' => $alert_bar ] );
```

File: src/Display.php [hunk 4]
```diff
@@ -90,7 +90,30 @@ public function add_body_class( $classes ) {
 	 */
 	protected function template( $part, $args = array() ) {
 		extract( $args );
```
Comment (rosswintle) [inline src/Display.php:92]: Code shouldn't `extract()`

https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#dont-extract

File: src/Display.php [hunk 5]
```diff
@@ -90,7 +90,30 @@ public function add_body_class( $classes ) {
 	 */
 	protected function template( $part, $args = array() ) {
```
Comment (rosswintle) [inline src/Display.php:91]: Short array syntax please
```suggestion
	protected function template( $part, $args = [] ) {
```

File: templates/alert-bar.php (hunk 3 above)
Comment (rosswintle) [reply templates/alert-bar.php:20]: Don't worry about this - keep the file consistent and we'll update them all in one go if/when we add coding standards.

File: src/Display.php (hunk 5 above)
Comment (rosswintle) [reply src/Display.php:91]: Don't worry about this - keep the file consistent and we'll update them all in one go if/when we add coding standards.

File: src/Display.php (hunk 4 above)
Comment (rosswintle) [reply src/Display.php:92]: Seems this is common in templates. Leave it for now. We'll clean up later.

File: src/Display.php (hunk 1 above)
Comment (sanjucta) [reply src/Display.php]: Done

**Review by rosswintle — APPROVED** [review]
Good to merge!

**Pull request merged**